### PR TITLE
Add DEStat

### PR DESCRIPTION
**Pull request by YingboMa**
```julia
julia> using OrdinaryDiffEq

julia> prob = ODEProblem((du,u,p,t)->@.(du=0.01*u), ones(100), (0,100.));

julia> solve(prob, Tsit5()).destats
DiffEqBase.DEStats
Number of function evaluations: 39
Number of W matrix evaluations: 0
Number of linear solves:        0
Number of accepted steps:       6
Number of rejected steps:       0

julia> solve(prob, AutoTsit5(Rosenbrock23())).destats
DiffEqBase.DEStats
Number of function evaluations: 39
Number of W matrix evaluations: 0
Number of linear solves:        0
Number of accepted steps:       6
Number of rejected steps:       0
Maximum eigenvalue recorded:    0.010000000023693143

julia> solve(prob, ImplicitEuler()).destats
DiffEqBase.DEStats
Number of function evaluations: 265
Number of W matrix evaluations: 64
Number of linear solves:        72
Number of accepted steps:       39
Number of rejected steps:       25
```

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
I don't think this counts the linear solves in a Rosenbrock method?

How is the number of function evaluations for AutoTsit5 so high there? It seems like it doesn't reset each time.

**Comment by ChrisRackauckas** [comment]
Fails on ParameterizedFunctions, that's tricky.

File: src/nlsolve/newton.jl [hunk 1]
```diff
@@ -54,6 +54,7 @@ Equations II, Springer Series in Computational Mathematics. ISBN
   else
     dz = _reshape(W \ _vec(ztmp), axes(ztmp))
   end
+  integrator.destats.nsolve += 1
```
Comment (kanav99) [inline src/nlsolve/newton.jl]: This is currently incrementing the linear solve counter on non-linear solves. I guess we should have two counters - each for linear and nonlinear.

Comment (kanav99) [reply src/nlsolve/newton.jl]: Oh my bad, still this doesn't counts the linear solves in the Rosenbrocks.

Comment (ChrisRackauckas) [reply src/nlsolve/newton.jl]: yeah, we can keep improving this

**Pull request merged**